tests/Unit/Domain/Creators/Test_ShapeMapOptions.cpp

```diff
@@ -113,11 +174,100 @@ void test_funcs() {
                                    DataVector{0.0}, DataVector{0.0}});
   }
+  {
+    const std::string test_filename{"TotalEclipseOfTheHeart.h5"};
+    if (file_system::check_if_file_exists(test_filename)) {
+      file_system::rm(test_filename, true);
+    }
+    const std::array<std::string, 3> subfile_names{"Ylm_coefs", "dt_Ylm_coefs",
+                                                   "d2t_Ylm_coefs"};
+    const double time = 1.7;
+    // Purposefully larger than the LMax in the options so that the
+    // Strahlkorpers will be restricted
+    const size_t file_l_max = 10;
+    std::uniform_real_distribution<double> distribution(0.1, 2.0);
+    using Frame = ::Frame::Distorted;
+    std::array<ylm::Strahlkorper<Frame>, 3> strahlkorpers{};
+    std::vector<std::string> legend{};
+    std::vector<double> data{};
+
+    // Scoped to close h5 file
+    {
+      h5::H5File<h5::AccessType::ReadWrite> test_file(test_filename, true);
+      for (size_t i = 0; i < 3; i++) {
+        legend.clear();
+        data.clear();
+        const auto radius = make_with_random_values<DataVector>(
+            generator, distribution,
+            DataVector(ylm::Spherepack::physical_size(file_l_max, file_l_max),
+                       std::numeric_limits<double>::signaling_NaN()));
+        gsl::at(strahlkorpers, i) = ylm::Strahlkorper<Frame>(
+            file_l_max, file_l_max, radius, std::array{0.0, 0.0, 0.0});
+
+        ylm::fill_ylm_legend_and_data(
+            make_not_null(&legend), make_not_null(&(data)),
+            gsl::at(strahlkorpers, i), time, file_l_max);
+
+        auto& file =
+            test_file.insert<h5::Dat>("/" + gsl::at(subfile_names, i), legend);
+        file.append(data);
+        test_file.close_current_object();
+      }
+    }
+
+    const auto shape_map_options = TestHelpers::test_creation<
+        domain::creators::time_dependent_options::ShapeMapOptions<
+            false, domain::ObjectLabel::None>>(
+        "LMax: 8\n"
+        "InitialValues:\n"
+        "  H5Filename: TotalEclipseOfTheHeart.h5\n"
+        "  SubfileNames:\n"
+        "    - Ylm_coefs\n"
+        "    - dt_Ylm_coefs\n"
+        "    - d2t_Ylm_coefs\n"
+        "  MatchTime: 1.7\n"
+        "  MatchTimeEpsilon: Auto\n"
+        "  SetL1CoefsToZero: True\n"
+        "SizeInitialValues: [1.1, 2.2, 3.3]");
+
+    const auto [shape_funcs, size_funcs] =
+        domain::creators::time_dependent_options::initial_shape_and_size_funcs(
+            shape_map_options, inner_radius);
+
+    ylm::SpherepackIterator iter{l_max, l_max};
+    ylm::SpherepackIterator file_iter{file_l_max, file_l_max};
+    for (size_t i = 0; i < shape_funcs.size(); i++) {
+      const size_t expected_size = ylm::Spherepack::spectral_size(l_max, l_max);
+      // Make sure they were restricted properly
+      CHECK(gsl::at(shape_funcs, i).size() == expected_size);
+
+      // Loop pointwise so we only check the coefficients that matter. Also
```

Comment:
Also what? :D

Reply:
Could you expand on what the confusion is? The coefficients as stored in spherepack have a ton of zeros in them which we don't need to check

Reply:
Oh sorry there's just an "Also" at the end of this line with nothing following it, sort of creating suspense :)

Reply:
Ohhhhhhhh. lol oops. I'll get that when I squash